src/main/scala/ai/privado/rulevalidator/YamlFileValidator.scala

```diff
@@ -0,0 +1,195 @@
+package ai.privado.rulevalidator
+
+import ai.privado.entrypoint.CommandConstants
+import ai.privado.model.CatLevelOne
+import better.files.File
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.networknt.schema.{JsonSchema, JsonSchemaFactory, SpecVersion, ValidationMessage}
+import org.slf4j.LoggerFactory
+
+import java.util
+import scala.jdk.CollectionConverters.SetHasAsScala
+import scala.io.Source
+
+
+object YamlFileValidator {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  private val PRETTY_LINE_SEPARATOR = "-"*100
+
+  private val SCHEMA_DIR_PATH = "/ai/privado/rulevalidator/schema/"
+  private val JSON_SCHEMA_VERSION = SpecVersion.VersionFlag.V7
+  private val RULES_FOLDER_IN_CONFIG_DIR = "rules"
```

Comment:
Along with validation on `rules`, also need to support validation on `config`. 
If possible let's add that as well, otherwise can take that as part of next iteration
cc - @pandurangpatil

Reply:
@pandurangpatil @HiteshMah-Jan  [open]

Reply:
We'll handle configs separately, I create another MR for it :)